packages/@react-native/tester/js/components/RNTesterPage.js

```diff
@@ -40,7 +41,7 @@ class RNTesterPage extends React.Component<Props> {
       <RNTesterThemeContext.Consumer>
         {theme => {
           return (
-            <View
+            <View testID={this.props.testID}
```

Comment:
I'm not sure this is the right scope to add this.

If we want to dump at "page" granularity it would be valuable to use a common test id. Imagine something like the "visitAllPages" test, where it visited every page and dumped "tester-page".

We will want to be able to write test cases at a finer granularity than the page though. This helps make the snapshot easier to decipher, and limits the effect of things like RNTester UI changes.

There has been some upstream work to add detox tests in "packages/rn-tester". This work has also included adding more test ids to tester examples. The granularity for the test ids they have is I think about what we want.

Reply:
That's a good point - do you think adding this prop to `RNTesterBlock` would be ok instead?

Reply:
Checked out the commits you mentioned, eg https://github.com/facebook/react-native/commit/d1f09f7390a89a951299b718e860ef0fdd3b8de9#
and it looks like they add the IDs on individual components, such as the Switch or Text. Is that what you're suggesting?

Reply:
Yeah, basically just the most scoped view per-individual example, instead of the entire tree.

Re adding to `RNTesterBlock`, I would probably avoid doing something on an interface used that often unless we also make the change upstream.

Reply:
Done